examples/profile/certificatehelper.go

```diff
@@ -122,22 +115,60 @@ func generateSelfSignedCertificate(certPath string, keyPath string, host string)
 		return err
 	}
 
+	err = createPemFile(certPath, derBytes)
+	if err != nil {
+		log.Printf("failed to create pem file at %q: %s", certPath, err)
+		return err
+	}
+	log.Printf("written %s\n", certPath)
+
+	err = createKeyFile(keyPath, priv)
+	if err != nil {
+		log.Printf("failed to create key file at %q: %s", keyPath, err)
+		return err
+	}
+	log.Printf("written %s\n", keyPath)
+
+	return nil
+}
+
+func createPemFile(certPath string, derBytes []byte) error {
+
 	certOut, err := os.Create(certPath)
+	defer certOut.Close()
```

Comment:
you should place this after the nil check (before `pem.Encode`), no point in closing when it wasn't opened (because of an error)